Examples/ObjectiveC/UserTrackingModesExample.m

```diff
@@ -3,23 +3,164 @@
 
+// Subclass UIButton to create a custom user tracking mode button
+@interface UserLocationButton : UIButton
+@property (nonatomic) CAShapeLayer *arrow;
+@end
+
+const CGFloat UserLocationButtonSize = 80;
+
+@implementation UserLocationButton
+
+- (instancetype)initWithButtonSize:(CGFloat)buttonSize {
+    if (self = [super init]) {
+        self.frame = CGRectMake(0, 0, buttonSize, buttonSize);
+        self.backgroundColor = [[UIColor whiteColor] colorWithAlphaComponent:0.8];
+        self.layer.cornerRadius = 4;
+        
+        CAShapeLayer *arrow = [[CAShapeLayer alloc] init];
+        
+        arrow.path = [self arrowPath];
+        arrow.lineWidth = 2;
+        arrow.lineJoin = kCALineJoinRound;
+        arrow.bounds = CGRectMake(0, 0, UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+        arrow.position = CGPointMake(UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+        arrow.shouldRasterize = YES;
+        arrow.rasterizationScale = [[UIScreen mainScreen] scale];
+        arrow.drawsAsynchronously = YES;
+        
+        self.arrow = arrow;
+        
+        [self updateArrow:MGLUserTrackingModeNone];
+        [self.layer addSublayer:self.arrow];
+    }
+    
+    return self;
+}
+
+- (CGPathRef) arrowPath {
+    
+    CGFloat max = UserLocationButtonSize / 2;
+    
+    UIBezierPath *bezierPath = [[UIBezierPath alloc] init];
+    
+    [bezierPath moveToPoint:CGPointMake(max * 0.5, 0)];
+    
+    [bezierPath addLineToPoint:CGPointMake(max * 0.1, max)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.5, max * 0.65)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.9, max)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.5, 0)];
+    [bezierPath closePath];
+    
+    return bezierPath.CGPath;
+}
+
+-(void)updateArrow:(MGLUserTrackingMode)mode {    
+    switch (mode) {
+        case MGLUserTrackingModeNone:
+            [self updateArrowStyle:[[UIColor whiteColor] CGColor]
+                     withFillColor:[[UIColor clearColor] CGColor]
+                       andRotation:0];
+            break;
+        case MGLUserTrackingModeFollow:
+            [self updateArrowStyle:self.tintColor.CGColor
+                     withFillColor:[[UIColor clearColor] CGColor]
+                       andRotation:0.66];
+            break;
+        case MGLUserTrackingModeFollowWithHeading:
+            [self updateArrowStyle:[[UIColor clearColor] CGColor]
+                     withFillColor:self.tintColor.CGColor
+                       andRotation:0.66];
+            break;
+        case MGLUserTrackingModeFollowWithCourse:
+            [self updateArrowStyle:[[UIColor clearColor] CGColor]
+                     withFillColor:self.tintColor.CGColor
+                       andRotation:0];
+            break;
+    }
+}
+
+-(void)updateArrowStyle:(CGColorRef)strokeColor withFillColor:(CGColorRef) fillColor andRotation:(CGFloat) rotation {
```

Comment:
I think we can get rid of the `with` and `and` here (though ObjC was once that verbose, this is largely a Swift thing now).

Reply:
`updateArrowStyle` doesn’t update a style, it updates the stroke color.